libs/platform/unit/platform_unit_test.cpp

```diff
@@ -703,6 +707,54 @@ BIT_MASK_TEST(65);
 BIT_MASK_TEST(129);
 BIT_MASK_TEST(1025);
 
+TEST_CASE("completion", "[platform]")
+{
+    _test_helper test_helper;
+
+    auto test = [](bool complete) {
+        struct _completion_context
+        {
+            ebpf_result_t result;
+        } completion_context = {EBPF_PENDING};
+
+        struct _cancellation_context
+        {
+            bool cancelled;
+        } cancellation_context = {false};
+
+        REQUIRE(
```

Comment:
trying to understand whose responsibility it would be to enter exit epoch. Because the epoch logic would ensure the tracker would be valid, if one thread (say initiator cancel) accesses it while the other (handler completes) deletes it.

Reply:
All calls to the async module must be within a epoch pair.